The pkg_info(1) behavior has changed in OpenBSD >7.3.
The error message ``Can't find`` should not lead to an error case
(https://github.com/ansible-collections/community.general/pull/6785)

Index: ansible_collections/community/general/plugins/modules/openbsd_pkg.py
--- ansible_collections/community/general/plugins/modules/openbsd_pkg.py.orig
+++ ansible_collections/community/general/plugins/modules/openbsd_pkg.py
@@ -169,7 +169,11 @@ def get_package_state(names, pkg_spec, module):
         rc, stdout, stderr = execute_command(command, module)
 
         if stderr:
-            module.fail_json(msg="failed in get_package_state(): " + stderr)
+            match = ("Can't find inst:%s" % name) in stderr
+            if match:
+                pkg_spec[name]['installed_state'] = False
+            else:
+                module.fail_json(msg="failed in get_package_state(): " + stderr)
 
         if stdout:
             # If the requested package name is just a stem, like "python", we may
